### PR TITLE
Add PHP 8.2 French release page

### PR DESCRIPTION
**Pull request by nlewo**
The french translation of the release page.

Instead of writing our own release notes French translation, at Gandi, we considered more useful to contribute it upstream.

Especially since your release note page looks pretty beautiful!

### CODE REVIEW

File: releases/8.2/languages/fr.php [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    'documentation' => 'Doc',
+    'main_title' => 'Released!',
+    'main_subtitle' => 'PHP 8.2 est une mise à jour majeure du langage PHP.<br class="display-none-md">Elle contient beaucoup de nouvelles fonctionnalités et d'optimisations, incluant les classes en lecture seule, les types "stand-alone" null, false, et true, la dépréciation des propriétés dynamiques, et plus encore.',
+    'upgrade_now' => 'Migrer à PHP 8.2 maintenant!',
+    'readonly_classes_title' => 'Classes en lecture seule',
+    'dnf_types_title' => 'Types DNF (forme normale disjonctive)',
+    'dnf_types_description' => 'Les types DNF permettent de combiner des types <a href="/manual/en/language.types.type-system.php#language.types.type-system.composite.union">union</a> et <a href="/manual/en/language.types.type-system.php#language.types.type-system.composite.intersection">intersection</a>, en suivant une règle stricte: lorsque des types union et intersection sont combinés, les types intersection doivent être groupés entre crochets.',
```
Comment (saundefined) [inline releases/8.2/languages/fr.php:11]: Thanks!

```suggestion
    'dnf_types_description' => 'Les types DNF permettent de combiner des types <a href="/manual/fr/language.types.type-system.php#language.types.type-system.composite.union">union</a> et <a href="/manual/fr/language.types.type-system.php#language.types.type-system.composite.intersection">intersection</a>, en suivant une règle stricte: lorsque des types union et intersection sont combinés, les types intersection doivent être groupés entre crochets.',
```

Aside: maybe it's more useful to use French doc links (here and below).

Comment (nlewo) [reply releases/8.2/languages/fr.php]: Yep, all doc links now points to the doc fr.

**Comment by nlewo** [comment]
The CI should be green now.

**Comment by saundefined** [comment]
@Girgias @devnexen @adoy can you review?

File: releases/8.2/languages/fr.php [hunk 2]
```diff
@@ -0,0 +1,40 @@
+<?php
+
+return [
+    'common_header' => 'PHP 8.2 est une mise à jour majeure du langage PHP. Classes en lecture seule, null, false, and true comme types "stand-alone", dépréciation des propriétés dynamiques, amélioration des performances et plus encore',
```
Comment (adoy) [inline releases/8.2/languages/fr.php:4]:
```suggestion
    'common_header' => 'PHP 8.2 est une mise à jour majeure du langage PHP. Classes en lecture seule, null, false, and true comme types "stand-alone", dépréciation des propriétés dynamiques, amélioration des performances et plus encore.',
```

File: releases/8.2/languages/fr.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    'documentation' => 'Doc',
+    'main_title' => 'Released!',
+    'main_subtitle' => 'PHP 8.2 est une mise à jour majeure du langage PHP.<br class="display-none-md">Elle contient beaucoup de nouvelles fonctionnalités et d\'optimisations, incluant les classes en lecture seule, les types "stand-alone" null, false, et true, la dépréciation des propriétés dynamiques, et plus encore.',
```
Comment (adoy) [inline releases/8.2/languages/fr.php:7]: French translation of `Deprecated` is `Obsolète`.

File: releases/8.2/languages/fr.php [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        <p>La classe <code>\Random\Randomizer</code> fournit une interface de haut niveau permettant, par exemple, de générer un entier aléatoire, mélanger un tableau ou une chaine de caractère, et plus encore.</p>',
+    'constants_in_traits_title' => 'Constantes dans les traits',
+    'constants_in_traits_description' => 'Il n\'est pas possible d\'accéder à une constante par le nom du trait, mais il est cependant possible d\'y accéder par la classe utilisant ce trait.',
+    'deprecate_dynamic_properties_title' => 'Dépréciation des propriétés dynamiques',
+    'deprecate_dynamic_properties_description' => '<p>Afin d\'éviter des erreurs, la création des propriétés dynamiques est dépréciée, sauf si la classe contient l\'attribut <code>#[\AllowDynamicProperties]</code>. <code>stdClass</code> autorise les propriétés dynamiques.</p>
+        <p>L\'utilisation des méthodes magiques <code>__get</code>/<code>__set</code> n\'est pas affectée par ce changement..</p>',
+    'new_classes_title' => 'Nouvelles classes, interfaces, et functions',
```
Comment (adoy) [inline releases/8.2/languages/fr.php:21]:
```suggestion
    'new_classes_title' => 'Nouvelles classes, interfaces, et fonctions',
```

File: releases/8.2/languages/fr.php [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    'bc_odbc' => 'Les extensions <code>ODBC</code> et <code>PDO_ODBC</code> échappent les noms d\'utilisateurs et mots de passe.',
+    'bc_str_locale_sensitive' => 'Les fonctions <a href="/manual/fr/function.strtolower.php"><code>strtolower</code></a> et <a href="/manual/fr/function.strtoupper.php"><code>strtoupper</code></a> ne sont plus sensibles à la locale.',
+    'bc_spl_enforces_signature' => 'Les methodes <a href="/manual/fr/splfileobject.getcsvcontrol.php"><code>SplFileObject::getCsvControl</code></a>, <a href="/manual/fr/splfileobject.fflush.php"><code>SplFileObject::fflush</code></a>, <a href="/manual/fr/splfileobject.ftell.php"><code>SplFileObject::ftell</code></a>, <a href="/manual/fr/splfileobject.fgetc.php"><code>SplFileObject::fgetc</code></a>, et <a href="/manual/fr/splfileobject.fpassthru.php"><code>SplFileObject::fpassthru</code></a> obligent leur signature.',
```
Comment (adoy) [inline releases/8.2/languages/fr.php:33]:
```suggestion
    'bc_spl_enforces_signature' => 'Les méthodes <a href="/manual/fr/splfileobject.getcsvcontrol.php"><code>SplFileObject::getCsvControl</code></a>, <a href="/manual/fr/splfileobject.fflush.php"><code>SplFileObject::fflush</code></a>, <a href="/manual/fr/splfileobject.ftell.php"><code>SplFileObject::ftell</code></a>, <a href="/manual/fr/splfileobject.fgetc.php"><code>SplFileObject::fgetc</code></a>, et <a href="/manual/fr/splfileobject.fpassthru.php"><code>SplFileObject::fpassthru</code></a> obligent leur signature.',
```

Comment (adoy) [inline releases/8.2/languages/fr.php:33]: I would say something like
`renforcent leur signature` more than `obligent leur signature`

File: releases/8.2/fr.php [hunk 6]
```diff
@@ -0,0 +1,5 @@
+<?php
+
+$lang = 'fr';
+
+include_once __DIR__ . '/release.inc';
```
Comment (Girgias) [inline releases/8.2/fr.php:5]: Nit: missing EOL before EOF.

File: releases/8.2/languages/fr.php [hunk 7]
```diff
@@ -0,0 +1,40 @@
+<?php
+
+return [
+    'common_header' => 'PHP 8.2 est une mise à jour majeure du langage PHP. Classes en lecture seule, null, false, and true comme types "stand-alone", dépréciation des propriétés dynamiques, amélioration des performances et plus encore',
+    'documentation' => 'Doc',
+    'main_title' => 'Released!',
+    'main_subtitle' => 'PHP 8.2 est une mise à jour majeure du langage PHP.<br class="display-none-md">Elle contient beaucoup de nouvelles fonctionnalités et d\'optimisations, incluant les classes en lecture seule, les types "stand-alone" null, false, et true, la dépréciation des propriétés dynamiques, et plus encore.',
+    'upgrade_now' => 'Migrer à PHP 8.2 maintenant!',
+    'readonly_classes_title' => 'Classes en lecture seule',
+    'dnf_types_title' => 'Types DNF (forme normale disjonctive)',
```
Comment (Girgias) [inline releases/8.2/languages/fr.php:10]:
```suggestion
    'dnf_types_title' => 'Types FDN (forme normale disjonctive)',
```

File: releases/8.2/languages/fr.php [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    'bc_spl_false' => 'La méthode <a href="/manual/fr/splfileobject.haschildren.php"><code>SplFileObject::hasChildren</code></a> a un type de retour "tentative" <code>false</code>.',
+    'bc_spl_null' => 'La méthode <a href="/manual/fr/splfileobject.getchildren.php"><code>SplFileObject::getChildren</code></a> a un type de retour "tentative" <code>null</code>.',
+    'bc_spl_deprecated' => 'La méthode interne <code>SplFileInfo::_bad_state_ex</code> a été dépréciée.',
+    'footer_title' => 'Meilleures performances, meilleure syntaxe et amélioration de la sureté du typage.',
+    'footer_description' => '<p>Pour le téléchargement des sources de PHP 8.2 veuillez visiter la page de  <a href="/downloads">téléchargement</a> page. Les binaires Windows peuvent être trouvés sur le site de <a href="https://windows.php.net/download">PHP Pour Windows</a> site. La liste des changements est notée dans le <a href="/ChangeLog-8.php#PHP_8_2">ChangeLog</a>.</p>
```
Comment (adoy) [inline releases/8.2/languages/fr.php:38]:
```suggestion
    'footer_description' => '<p>Pour le téléchargement des sources de PHP 8.2 veuillez visiter la page de <a href="/downloads">téléchargement</a>. Les binaires Windows peuvent être trouvés sur le site de <a href="https://windows.php.net/download">PHP Pour Windows</a>. La liste des changements est notée dans le <a href="/ChangeLog-8.php#PHP_8_2">ChangeLog</a>.</p>
```

File: releases/8.2/languages/fr.php [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+    'documentation' => 'Doc',
+    'main_title' => 'Released!',
+    'main_subtitle' => 'PHP 8.2 est une mise à jour majeure du langage PHP.<br class="display-none-md">Elle contient beaucoup de nouvelles fonctionnalités et d\'optimisations, incluant les classes en lecture seule, les types "stand-alone" null, false, et true, la dépréciation des propriétés dynamiques, et plus encore.',
+    'upgrade_now' => 'Migrer à PHP 8.2 maintenant!',
+    'readonly_classes_title' => 'Classes en lecture seule',
+    'dnf_types_title' => 'Types DNF (forme normale disjonctive)',
+    'dnf_types_description' => 'Les types DNF permettent de combiner des types <a href="/manual/fr/language.types.type-system.php#language.types.type-system.composite.union">union</a> et <a href="/manual/fr/language.types.type-system.php#language.types.type-system.composite.intersection">intersection</a>, en suivant une règle stricte: lorsque des types union et intersection sont combinés, les types intersection doivent être groupés entre crochets.',
```
Comment (Girgias) [inline releases/8.2/languages/fr.php:11]:
```suggestion
    'dnf_types_description' => 'Les types FDN permettent de combiner des types <a href="/manual/fr/language.types.type-system.php#language.types.type-system.composite.union">union</a> et <a href="/manual/fr/language.types.type-system.php#language.types.type-system.composite.intersection">intersection</a>, en suivant une règle stricte: lorsque des types union et intersection sont combinés, les types intersection doivent être groupés entre crochets.',
```

File: releases/8.2/languages/fr.php [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+    'bc_spl_false' => 'La méthode <a href="/manual/fr/splfileobject.haschildren.php"><code>SplFileObject::hasChildren</code></a> a un type de retour "tentative" <code>false</code>.',
+    'bc_spl_null' => 'La méthode <a href="/manual/fr/splfileobject.getchildren.php"><code>SplFileObject::getChildren</code></a> a un type de retour "tentative" <code>null</code>.',
+    'bc_spl_deprecated' => 'La méthode interne <code>SplFileInfo::_bad_state_ex</code> a été dépréciée.',
+    'footer_title' => 'Meilleures performances, meilleure syntaxe et amélioration de la sureté du typage.',
+    'footer_description' => '<p>Pour le téléchargement des sources de PHP 8.2 veuillez visiter la page de  <a href="/downloads">téléchargement</a> page. Les binaires Windows peuvent être trouvés sur le site de <a href="https://windows.php.net/download">PHP Pour Windows</a> site. La liste des changements est notée dans le <a href="/ChangeLog-8.php#PHP_8_2">ChangeLog</a>.</p>
+        <p>Le <a href="/manual/fr/migration82.php">guide de migration</a> est disponible dans le manuel PHP. Veuillez le consulter pour une liste détailler des nouvelles fonctionnalités et changements non rétrocompatibles.</p>',
```
Comment (adoy) [inline releases/8.2/languages/fr.php:39]:
```suggestion
        <p>Le <a href="/manual/fr/migration82.php">guide de migration</a> est disponible dans le manuel PHP. Veuillez le consulter pour une liste détaillée des nouvelles fonctionnalités et changements non rétrocompatibles.</p>',
```

File: releases/8.2/languages/fr.php [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+    'dnf_types_title' => 'Types DNF (forme normale disjonctive)',
+    'dnf_types_description' => 'Les types DNF permettent de combiner des types <a href="/manual/fr/language.types.type-system.php#language.types.type-system.composite.union">union</a> et <a href="/manual/fr/language.types.type-system.php#language.types.type-system.composite.intersection">intersection</a>, en suivant une règle stricte: lorsque des types union et intersection sont combinés, les types intersection doivent être groupés entre crochets.',
+    'null_false_true_types_title' => 'Permettre <code>null</code>, <code>false</code>, et <code>true</code> comme types stand-alone',
+    'random_title' => 'Nouvelle extension "Random"',
+    'random_description' => '<p>L\'extension "random" fournit une nouvelle API orientée objet de génération de nombres aléatoires. Plutôt que de reposer sur un générateur de nombres aléatoires (utilisant l\'algorithme Mersenne Twister) globalement initialisé, l\'API orientée objet offre plusieurs classes ("Engine"s) permettant d\'accéder à des algorithmes modernes stockant leur état au sein des objets, afin de fournir des séquences d\'initialisations bien distinctes.</p>
```
Comment (Girgias) [inline releases/8.2/languages/fr.php:14]:
```suggestion
    'random_description' => '<p>L\'extension "random" fournit une nouvelle API orientée objet de génération de nombres aléatoires. Plutôt que de reposer sur un générateur de nombres aléatoires (utilisant l\'algorithme Mersenne Twister) globalement initialisé, l\'API orientée objet offre plusieurs classes ("Moteurs") permettant d\'accéder à des algorithmes modernes stockant leur état au sein des objets, afin de fournir des séquences d\'initialisations bien distinctes.</p>
```

File: releases/8.2/languages/fr.php [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+    'bc_title' => 'Dépréciations et changements non-retrocompatibles',
+    'bc_string_interpolation' => 'Dépréciation de l\'interpolation <code>${}</code>.',
+    'bc_utf8' => 'Dépréciation des fonctions <a href="/manual/fr/function.utf8-encode.php"><code>utf8_encode</code></a> et <a href="/manual/fr/function.utf8-decode.php"><code>utf8_decode</code></a>.',
+    'bc_datetime' => 'Les méthodes <a href="/manual/fr/datetime.createfromimmutable.php"><code>DateTime::createFromImmutable</code></a> et <a href="/manual/fr/datetimeimmutable.createfrommutable.php"><code>DateTimeImmutable::createFromMutable</code></a> ont comme type de retour "tentative" <code>static</code>.',
```
Comment (Girgias) [inline releases/8.2/languages/fr.php:30]: ``tentative`` seems kinda wrong, Word Reference is suggesting something like "provisoire" or "de principe"

**Review by Girgias — COMMENTED** [review]
Couple of comments but looks good overall.

File: releases/8.2/languages/fr.php (hunk 11 above)
Comment (nlewo) [reply releases/8.2/languages/fr.php:14]: I was thinking the name of the class is `Engine`, that's why i preserved the english word.

File: releases/8.2/languages/fr.php (hunk 2 above)
Comment (nlewo) [reply releases/8.2/languages/fr.php:4]: Applied, but note the english version doesn't end with a dot.

File: releases/8.2/languages/fr.php (hunk 12 above)
Comment (nlewo) [reply releases/8.2/languages/fr.php]: Yes, i really didn't know how to translate "tentative" :/

Let's then use "provisoire".

File: releases/8.2/languages/fr.php (hunk 5 above)
Comment (nlewo) [reply releases/8.2/languages/fr.php]: done

File: releases/8.2/languages/fr.php (hunk 8 above)
Comment (nlewo) [reply releases/8.2/languages/fr.php]: done

File: releases/8.2/languages/fr.php (hunk 5 above)
Comment (nlewo) [reply releases/8.2/languages/fr.php]: applied

File: releases/8.2/fr.php (hunk 6 above)
Comment (nlewo) [reply releases/8.2/fr.php]: done.

File: releases/8.2/languages/fr.php (hunk 3 above)
Comment (nlewo) [reply releases/8.2/languages/fr.php]: Updated to "désormais obsolète"

File: releases/8.2/languages/fr.php (hunk 10 above)
Comment (nlewo) [reply releases/8.2/languages/fr.php]: I also fixed the source of my copy/paste (8.0 release notes) :)

**Review by nlewo — COMMENTED** [review]
Thanks for all your comments!

File: releases/8.2/languages/fr.php (hunk 11 above)
Comment (nlewo) [reply releases/8.2/languages/fr.php:14]: @Girgias are you sure i need to change it to "Moteurs"?

File: releases/8.2/languages/fr.php (hunk 10 above)
Comment (adoy) [reply releases/8.2/languages/fr.php]: Nice catch. Thanks :-)

File: releases/8.2/languages/fr.php (hunk 3 above)
Comment (adoy) [reply releases/8.2/languages/fr.php]: Thanks @nlewo, I only reported the problem once, but it is repeated several times in the file.

File: releases/8.2/languages/fr.php [hunk 13]
```diff
@@ -0,0 +1,40 @@
+<?php
+
+return [
+    'common_header' => 'PHP 8.2 est une mise à jour majeure du langage PHP. Classes en lecture seule, null, false, and true comme types "stand-alone", propriétés dynamiques désormais obsolètes, amélioration des performances et plus encore.',
+    'documentation' => 'Doc',
+    'main_title' => 'Released!',
+    'main_subtitle' => 'PHP 8.2 est une mise à jour majeure du langage PHP.<br class="display-none-md">Elle contient beaucoup de nouvelles fonctionnalités et d\'optimisations, incluant les classes en lecture seule, les types "stand-alone" null, false, et true, les propriétés dynamiques désormais obsolètes, et plus encore.',
+    'upgrade_now' => 'Migrer à PHP 8.2 maintenant!',
```
Comment (devnexen) [inline releases/8.2/languages/fr.php:8]: I would be more comfortable with
`Migrer vers PHP 8.2 maintenant` or similar

Comment (nlewo) [reply releases/8.2/languages/fr.php]: Agreed.

File: releases/8.2/languages/fr.php [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+    'new_reflection' => 'Nouvelles méthodes <a href="/manual/fr/reflectionfunction.isanonymous.php"><code>ReflectionFunction::isAnonymous</code></a> et <a href="/manual/fr/reflectionmethod.hasprototype.php"><code>ReflectionMethod::hasPrototype</code></a>.',
+    'new_functions' => 'Nouvelles fonctions <a href="/manual/fr/function.curl_upkeep.php"><code>curl_upkeep</code></a>, <a href="/manual/fr/function.memory-reset-peak-usage.php"><code>memory_reset_peak_usage</code></a>, <a href="/manual/fr/function.ini-parse-quantity.php"><code>ini_parse_quantity</code></a>, <a href="/manual/fr/function.libxml-get-external-entity-loader.php"><code>libxml_get_external_entity_loader</code></a>, <a href="/manual/fr/function.sodium-crypto-stream-xchacha20-xor-ic.php"><code>sodium_crypto_stream_xchacha20_xor_ic</code></a>, <a href="/manual/fr/function.openssl-cipher-key-length.php"><code>openssl_cipher_key_length</code></a>.',
+    'bc_title' => 'Dépréciations et changements non-retrocompatibles',
```
[the remark this answers is not in the document]
Comment (nlewo) [reply releases/8.2/languages/fr.php]: done.

File: releases/8.2/languages/fr.php [hunk 15]
```diff
@@ -0,0 +1,40 @@
+<?php
+
+return [
+    'common_header' => 'PHP 8.2 est une mise à jour majeure du langage PHP. Classes en lecture seule, null, false, and true comme types "stand-alone", propriétés dynamiques désormais obsolètes, amélioration des performances et plus encore.',
```
Comment (adoy) [inline releases/8.2/languages/fr.php:4]:
```suggestion
    'common_header' => 'PHP 8.2 est une mise à jour majeure du langage PHP. Classes en lecture seule, null, false, et true comme types "stand-alone", propriétés dynamiques désormais obsolètes, amélioration des performances et plus encore.',
```

**Comment by adoy** [comment]
Beau travail !

File: releases/8.2/languages/fr.php (hunk 15 above)
Comment (nlewo) [reply releases/8.2/languages/fr.php]: Fixed.

**Review by devnexen — APPROVED** [review]
Looks good to me

File: releases/8.2/languages/fr.php (hunk 11 above)
Comment (Girgias) [reply releases/8.2/languages/fr.php:14]: The wording of the English doesn't indicate to me that these are class names. But if one should keep the class name something like:

> [l'API OO] offres plusieurs classes Random\Engine ("Moteurs") permettant [...]

But frankly, the English looks a bit wonky now that I read it.

File: releases/8.2/languages/fr.php [hunk 16]
```diff
@@ -0,0 +1,40 @@
...
+    'documentation' => 'Doc',
+    'main_title' => 'Released!',
+    'main_subtitle' => 'PHP 8.2 est une mise à jour majeure du langage PHP.<br class="display-none-md">Elle contient beaucoup de nouvelles fonctionnalités et d\'optimisations, incluant les classes en lecture seule, les types "stand-alone" null, false, et true, les propriétés dynamiques désormais obsolètes, et plus encore.',
+    'upgrade_now' => 'Migrer vers PHP 8.2 maintenant!',
+    'readonly_classes_title' => 'Classes en lecture seule',
+    'dnf_types_title' => 'Types FDN (forme normale disjonctive)',
+    'dnf_types_description' => 'Les types FDNpermettent de combiner des types <a href="/manual/fr/language.types.type-system.php#language.types.type-system.composite.union">union</a> et <a href="/manual/fr/language.types.type-system.php#language.types.type-system.composite.intersection">intersection</a>, en suivant une règle stricte: lorsque des types union et intersection sont combinés, les types intersection doivent être groupés entre crochets.',
```
Comment (Girgias) [inline releases/8.2/languages/fr.php:11]:
```suggestion
    'dnf_types_description' => 'Les types FDN permettent de combiner des types <a href="/manual/fr/language.types.type-system.php#language.types.type-system.composite.union">union</a> et <a href="/manual/fr/language.types.type-system.php#language.types.type-system.composite.intersection">intersection</a>, en suivant une règle stricte: lorsque des types union et intersection sont combinés, les types intersection doivent être groupés entre crochets.',
```

Comment (nlewo) [reply releases/8.2/languages/fr.php]: Fixed.

File: releases/8.2/languages/fr.php (hunk 11 above)
Comment (nlewo) [reply releases/8.2/languages/fr.php]: So, i propose to simply remove the stuffs between parenthesis!
Is it ok for you?

Comment (Girgias) [reply releases/8.2/languages/fr.php]: Sure, fine by me. :-)

Comment (nlewo) [reply releases/8.2/languages/fr.php]: @Girgias thx! Done.

**Pull request merged**